From 57c37232ca33c09c68779c1987528928f8ce585b Mon Sep 17 00:00:00 2001
From: "haiyang.ren" <haiyang.ren@amlogic.com>
Date: Thu, 6 Jun 2024 09:40:47 +0000
Subject: [PATCH 7/9] audio: headphone silence issue on t3x [2/2]

PD#SWPL-170310

Problem:
1. No headphone detection;
2. The t3x json configuration is incorrect;

Solution:
1. Add headphone detection;
2. Add t3x json configuration;

Verify:
T3x_BC302

Change-Id: I36f3306a2b4c0248ba7f31416cfe15692faec889
Signed-off-by: haiyang.ren <haiyang.ren@amlogic.com>
---
 .../aml-audio-hal/aml-audio-hal_git.bb        |  4 ++
 .../files/aml_audio_config.t3x.json           | 33 +++++++++++++++++
 .../aml-audio-hal/files/mixer_paths.t3x.xml   | 37 +++++++++++++++++++
 3 files changed, 74 insertions(+)
 create mode 100644 recipes-multimedia/aml-audio-hal/files/aml_audio_config.t3x.json
 create mode 100644 recipes-multimedia/aml-audio-hal/files/mixer_paths.t3x.xml

diff --git a/recipes-multimedia/aml-audio-hal/aml-audio-hal_git.bb b/recipes-multimedia/aml-audio-hal/aml-audio-hal_git.bb
index 16935a5d..e9f401ac 100644
--- a/recipes-multimedia/aml-audio-hal/aml-audio-hal_git.bb
+++ b/recipes-multimedia/aml-audio-hal/aml-audio-hal_git.bb
@@ -48,11 +48,13 @@ SRC_URI  += "\
   file://aml_audio_config.bm201.json \
   file://aml_audio_config.u212.json \
   file://aml_audio_config.tv.json \
+  file://aml_audio_config.t3x.json \
   file://aml_audio_config.s1a.json \
   file://mixer_paths.xml \
   file://mixer_paths.at301.xml \
   file://mixer_paths.t7.xml \
   file://mixer_paths.t5d.xml \
+  file://mixer_paths.t3x.xml \
   file://audio_hal_delay_base.json \
   file://audio_hal_delay_base.at301.json \
   file://audio_hal_delay_base.am301.json \
@@ -70,6 +72,7 @@ PROPERTY_SET_CONF:am301 = "aml_audio_config.am301.json"
 PROPERTY_SET_CONF:at301 = "aml_audio_config.at301.json"
 PROPERTY_SET_CONF:ap222 = "aml_audio_config.ap222.json"
 PROPERTY_SET_CONF:t7 = "aml_audio_config.tv.json"
+PROPERTY_SET_CONF:t3x = "aml_audio_config.t3x.json"
 PROPERTY_SET_CONF:bh201 = "aml_audio_config.bh201.json"
 PROPERTY_SET_CONF:bm201 = "aml_audio_config.bm201.json"
 PROPERTY_SET_CONF:s1a = "aml_audio_config.s1a.json"
@@ -78,6 +81,7 @@ PROPERTY_SET_MIXER = "mixer_paths.xml"
 PROPERTY_SET_MIXER:at301 = "mixer_paths.at301.xml"
 PROPERTY_SET_MIXER:t7 = "mixer_paths.t7.xml"
 PROPERTY_SET_MIXER:t5d = "mixer_paths.t5d.xml"
+PROPERTY_SET_MIXER_t3x = "mixer_paths.t3x.xml"
 PROPERTY_SET_AVSYNC = "audio_hal_delay_base.json"
 PROPERTY_SET_AVSYNC:at301 = "audio_hal_delay_base.at301.json"
 PROPERTY_SET_AVSYNC:am301 = "audio_hal_delay_base.am301.json"
diff --git a/recipes-multimedia/aml-audio-hal/files/aml_audio_config.t3x.json b/recipes-multimedia/aml-audio-hal/files/aml_audio_config.t3x.json
new file mode 100644
index 00000000..53f63d9c
--- /dev/null
+++ b/recipes-multimedia/aml-audio-hal/files/aml_audio_config.t3x.json
@@ -0,0 +1,33 @@
+{
+	"TV_Platform": 1,
+	"STB_Platform": 0,
+	"Dual_Spdif_Support": 1,
+	"Ms12_Force_Ddp_Out": 0,
+	"Dtv_Audio_Skipamadec": 0,
+	"Dtv_Audio_Syncenable": 1,
+	"Codec_Support_List":[
+		{"Name": "PCM", "Format": "AUDIO_FORMAT_PCM", "Support": true, "Description": "pcm", "License":false},
+		{"Name": "MP3", "Format": "AUDIO_FORMAT_MP3", "Support": true, "Description": "MPEG-1/2 Layer1/Layer2/Layer3", "License":false},
+		{"Name": "AMRNB", "Format": "AUDIO_FORMAT_AMR_NB", "Support": false, "Description": "none", "License":false},
+		{"Name": "AMRWB", "Format": "AUDIO_FORMAT_AMR_WB", "Support": false, "Description": "none", "License":false},
+		{"Name": "AAC", "Format": "AUDIO_FORMAT_AAC", "Support": true, "Description": "none", "License":false},
+		{"Name": "HEAACv1", "Format": "AUDIO_FORMAT_HE_AAC_V1", "Support": true, "Description": "none", "License":false},
+		{"Name": "HEAACv2", "Format": "AUDIO_FORMAT_HE_AAC_V2", "Support": true, "Description": "none", "License":false},
+		{"Name": "VORBIS", "Format": "AUDIO_FORMAT_VORBIS", "Support": true, "Description": "none", "License":false},
+		{"Name": "OPUS", "Format": "AUDIO_FORMAT_OPUS", "Support": true, "Description": "none", "License":false},
+		{"Name": "DD", "Format": "AUDIO_FORMAT_AC3", "Support": true, "Description": "none", "License":true},
+		{"Name": "DDP", "Format": "AUDIO_FORMAT_E_AC3", "Support": true, "Description": "dolby digital plus", "License":true},
+		{"Name": "DTS", "Format": "AUDIO_FORMAT_DTS", "Support": true, "Description": "none", "License":true},
+		{"Name": "DTSHD", "Format": "AUDIO_FORMAT_DTS_HD", "Support": true, "Description": "none", "License":true},
+		{"Name": "TRUEHD", "Format": "AUDIO_FORMAT_DOLBY_TRUEHD", "Support": true, "Description": "none", "License":true},
+		{"Name": "AACADIF", "Format": "AUDIO_FORMAT_AAC_ADIF", "Support": true, "Description": "none", "License":false},
+		{"Name": "WMA", "Format": "AUDIO_FORMAT_WMA", "Support": true, "Description": "WMA (v7,v8,v9)", "License":true},
+		{"Name": "WMAPRO", "Format": "AUDIO_FORMAT_WMA_PRO", "Support": true, "Description": "WMA 9 Pro/WMA 10 Pro", "License":true},
+		{"Name": "MP2", "Format": "AUDIO_FORMAT_MP2", "Support": true, "Description": "none", "License":false},
+		{"Name": "FLAC", "Format": "AUDIO_FORMAT_FLAC", "Support": true, "Description": "none", "License":false},
+		{"Name": "ALAC", "Format": "AUDIO_FORMAT_ALAC", "Support": true, "Description": "none", "License":true},
+		{"Name": "APE", "Format": "AUDIO_FORMAT_APE", "Support": true, "Description": "none", "License":true},
+		{"Name": "AC4", "Format": "AUDIO_FORMAT_AC4", "Support": true, "Description": "none", "License":true},
+		{"Name": "MAT", "Format": "AUDIO_FORMAT_MAT", "Support": true, "Description": "none", "License":true}
+	]
+}
\ No newline at end of file
diff --git a/recipes-multimedia/aml-audio-hal/files/mixer_paths.t3x.xml b/recipes-multimedia/aml-audio-hal/files/mixer_paths.t3x.xml
new file mode 100644
index 00000000..5208aca0
--- /dev/null
+++ b/recipes-multimedia/aml-audio-hal/files/mixer_paths.t3x.xml
@@ -0,0 +1,37 @@
+<mixer>
+  <ctl name="HDMIIN Audio output mode" value="I2S" />
+  <ctl name="HDMI ARC Switch" value="0" />
+
+  <path name="speaker">
+    <ctl name="AMP Maser Volume Mute" value="0" />
+  </path>
+
+  <path name="speaker_off">
+    <ctl name="AMP Maser Volume Mute" value="1" />
+  </path>
+
+  <path name="hdmi_arc">
+   <ctl name="HDMI ARC Switch" value="1" />
+  </path>
+
+  <path name="hdmi_arc_off">
+   <ctl name="HDMI ARC Switch" value="0" />
+  </path>
+
+  <path name="headphone">
+  </path>
+
+  <path name="main_mic">
+  </path>
+
+  <path name="headset-mic">
+  </path>
+
+  <path name="line_in">
+    <ctl name="TDMIN_B source select" value="acodec_adc" />
+  </path>
+
+  <path name="hdmirx_in">
+    <ctl name="TDMIN_B source select" value="hdmirx" />
+  </path>
+</mixer>
\ No newline at end of file
-- 
2.25.1

